services/visual_recognition/v3.js

```diff
@@ -62,301 +37,331 @@ module.exports = function (RED) {
+    node.service.listClassifiers(params, function(err, body) {
+      node.status({});
+      if (err) {
+        node.status({fill:'red', shape:'ring', 
+          text:'Delete All : call to listClassifiers failed'});
+        node.error(err, msg);   
+      } else {
+        // Array to hold async tasks
+        var asyncTasks=[], nbTodelete=0, nbdeleted=0;
+        nbTodelete = body.classifiers.length;
+        body.classifiers.forEach(function (aClassifier) {
+          asyncTasks.push(function (cb) {
+            var parms = {};
+            parms.classifier_id=aClassifier.classifier_id;
+            node.service.deleteClassifier(parms, function(err, body) {
+              if (err) {
+                node.error(err, msg);
+                //console.log('Error with the removal of classifier_id ' +
+                  //parms.classifier_id +' : ' + err);
+                return cb('error');
+              }
+              //console.log('Classifier ID '+ aClassifier.classifier_id + 
+              //  ' deleted successfully.');
+              nbdeleted++;
+              cb(null,parms.classifier_id);
+            });  
           });
-        }); // temp
-      } else if (feature==='createClassifier') {   
-          var listParams = {};
-          var asyncTasks = [];
-          var prop = null;
-          for (var k in msg.params)
-          {
-            prop = k;
-            if (prop.indexOf('_examples')>=0)
-            {
-              // before pushing the function into the task array wrap the push 
-              // in an IIFE function, passing in the 'prop' parameter
-              (function(prop, listParams, msg) {
+        });      
+        async.parallel(asyncTasks, function(error, deletedList){
+          if (deletedList.length===nbTodelete) {
+            msg.payload='see msg.result.error';
+            msg.result = 'All custom classifiers have been deleted.';
+          } else {
+            msg.payload='see msg.result.error';
+            msg.result = 'Some Classifiers could have not been deleted;'+
+            'See log for errors.';
+          }
+          node.send(msg);
+          node.status({});           
+        });
+      }
+    }); 
+  }  // delete all func 
+
+  function executeService(feature, params, node, msg) {
+    switch(feature) {
+    case 'classifyImage':
+      prepareParamsCommon(params, node, msg, function () {
+        node.service.classify(params, function(err, body, other) {
+          processResponse(err,body,feature,node,msg);
+        });
+      });
+      break;
+    case 'detectFaces':
+      prepareParamsCommon(params, node, msg, function () {
+        node.service.detectFaces(params, function(err, body, other) {
+          processResponse(err,body,feature,node,msg);
+        });
+      });
+      break;
+    case 'recognizeText':
+      prepareParamsCommon(params, node, msg, function () {
+        node.service.recognizeText(params, function(err, body, other) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['other' is defined but never used.](https://www.codacy.com/app/node-red-nodes-watson/node-red-node-watson/file/2737047197/issues/source?bid=3225115&fileBranchId=3352493#l282)